File: GameEngine_PC/commons/src/main/java/com/dferreira/commons/Vector3f.java
package com.dferreira.commons;

/**
 * Represents a 3D Vector with their x,y,z coordinates
 */
public class Vector3f extends Vector2f {

	/* The coordinates them selves */
	public float z;

	/**
	 * The constructor of the vector
	 * 
	 * @param x
	 *            coordinate
	 * @param y
	 *            coordinate
	 * @param z
	 *            coordinate
	 */
	public Vector3f(float x, float y, float z) {
		super(x, y);
		this.z = z;
	}

	/**
	 * Transforms a 3F vector into a string
	 */
	@Override
	public String toString() {
		return String.format("(%f,%f,%f)", x, y, z);
	}

	/**
	 * Normalizes the components of the vector
	 */
	public void normalise() {
		double magnitude = Math.sqrt((this.x * this.x) + (this.y * this.y) + (this.z * this.z));
		this.x = (float) (x / magnitude);
		this.y = (float) (y / magnitude);
		this.z = (float) (z / magnitude);
	}
}

File: GameEngine_IOS/GameEngine/Commons/Vector3f.swift
import Foundation

/**
 * Represents a 3D Vector with their x,y,z coordinates
 */
public class Vector3f : Vector2f {
    
    var z : Float;
    
    /**
     * The Initialize of the vector
     *
     * @param x
     *            X-coordinate
     * @param y
     *            Y-coordinate
     * @param z
     *            Z-coordinate
     */
    public init(x : Float , y : Float, z : Float) {
        self.z = z;
        super.init(x: x, y: y);
    }
    
    /**
     * Transforms a 3D vector into a string
     */
    var description : String {
        return String(format: "(%f,%f,%f)", x, y, z)
    }
    
    /**
     * Normalizes the components of the vector
     */
    public func normalise() {
        let magnitude : Float = Math.sqrt((self.x * self.x) + (self.y * self.y) + (self.z * self.z));
        self.x = (x / magnitude);
        self.y = (y / magnitude);
        self.z = (z / magnitude);
    }
}
